query-engine/connectors/sql-query-connector/src/value_ext.rs

```diff
@@ -0,0 +1,72 @@
+pub trait IntoTypedJsonExtension {
+    // Returns the type-name
+    fn type_name(&self, backward_compatible: bool) -> String;
+    /// Decorate all values with type-hints
+    fn as_typed_json(self, backward_compatible: bool) -> serde_json::Value;
+}
+
+impl<'a> IntoTypedJsonExtension for quaint::Value<'a> {
+    fn type_name(&self, backward_compatible: bool) -> String {
+        if self.is_null() {
+            return "null".to_owned();
+        }
+
+        let type_name = if backward_compatible {
+            match self {
+                quaint::Value::Boolean(_) => "bool",
+                quaint::Value::Int32(_) => "int",
+                quaint::Value::Int64(_) => "int",
+                quaint::Value::Float(_) => "float",
+                quaint::Value::Double(_) => "double",
+                quaint::Value::Json(_) => "json",
+                quaint::Value::Numeric(_) => "float",
+                quaint::Value::Array(_) => "array",
+                _ => "string",
+            }
+        } else {
+            match self {
+                quaint::Value::Int32(_) => "int",
+                quaint::Value::Int64(_) => "bigint",
+                quaint::Value::Float(_) => "float",
+                quaint::Value::Double(_) => "double",
+                quaint::Value::Text(_) => "string",
+                quaint::Value::Enum(_) => "enum",
+                quaint::Value::Bytes(_) => "bytes",
+                quaint::Value::Boolean(_) => "bool",
+                quaint::Value::Char(_) => "char",
+                quaint::Value::Numeric(_) => "decimal",
+                quaint::Value::Json(_) => "json",
+                quaint::Value::Xml(_) => "xml",
+                quaint::Value::Uuid(_) => "uuid",
+                quaint::Value::DateTime(_) => "datetime",
+                quaint::Value::Date(_) => "date",
+                quaint::Value::Time(_) => "time",
+                quaint::Value::Array(_) => "array",
```

Comment:
A lot of those types cannot be properly deserialized in JS. eg: `float`, `double`, `char`, `xml`, `uuid`, `datetime`, `date`, `time`. There are two reasons I thought we should retain the correct type information anyway:
1. Other clients _might_ be able to properly coerce those types (even though we don't plan on officially supporting more languages atm)
2. The JS client might want to properly coerce some of those types in the future (eg: `uuid`)

For now, though, the JS client should just leave those values untouched.